examples/tv-casting-app/android/App/app/src/main/jni/com/chip/casting/NsdDiscoveryListener.java

```diff
@@ -58,24 +59,33 @@ public void onDiscoveryStarted(String regType) {
 
   @Override
   public void onServiceFound(NsdServiceInfo service) {
-    Log.d(TAG, "Service discovery success. " + service);
-    if (service.getServiceType().equals(targetServiceType)) {
-      if (nsdManagerResolverAvailState != null) {
-        nsdManagerResolverAvailState.acquireResolver();
-      }
-      nsdManager.resolveService(
-          service,
-          new NsdResolveListener(
-              nsdManager,
-              deviceTypeFilter,
-              preCommissionedVideoPlayers,
-              successCallback,
-              failureCallback,
-              nsdManagerResolverAvailState,
-              1));
-    } else {
-      Log.d(TAG, "Ignoring discovered service: " + service.toString());
-    }
+    Executors.newSingleThreadExecutor()
```

Comment:
is it efficient to create a new Executor each time rather than running this in an executor belonging to the class or instance?

Reply:
Makes sense. Done!